docs/DeveloperGuide.adoc

```diff
@@ -319,6 +319,38 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+=== List feature
+==== Current Implementation
+
+The list mechanism is facilitated by `ModulePlanner`. `ModulePlanner` stores a list of all `Semester`-s and each `Semester` stores a list `modulesTaken` containing modules that the user has taken or is planning to take.
+It implements the following operation:
+
+* `ModulePlanner#listModulesTaken(int index)` -- Retrieves a list of modules from the `Semester` specified by `index`.
+
+The operation is exposed in `Model` interface as `Model#getFilteredTakenModuleList(int index)`.
+
+[NOTE]
+A valid index should be an integer between 1 to 8 inclusive, where index 1 represents year 1 semester 1, index 2 represents year 1 semester 2, and so on.
```

Comment:
Now some people may attribute the increasing `index` with the semester increasing only. I think we should add another example: `index 3 represents year 2 semester 1` to show the wrap around